algorithms/src/snark/marlin/marlin.rs

```diff
@@ -469,41 +502,42 @@ where
             .chain(second_oracles.iter())
             .chain(third_oracles.iter())
             .chain(fourth_oracles.iter())
+            .chain(fifth_oracles.iter())
             .collect();
         assert!(
             polynomials.len()
                 == keys_to_constraints.len() * 12 + // polys for row, col, rowcol, val
```

Comment:
Let's make a variable `num_unique_circuits = keys_to_constraints.len()`, and use that for these assertions. Slightly clearer IMO

Reply:
00c681431

Yes, in a future PR I'll also make a `num_index_oracles` function for consistency